// @dhuntleypro/afm-library/src/utils/constants.tsx




export const CONSTANTS = {
    appName: "",
    holderImageWelcome: 'https://appsformankind-assets.s3.amazonaws.com/Collections/Furniture/Furniture_1.jpg', 
    holderImageProductCard: 'https://appsformankind-assets.s3.amazonaws.com/Store/Jays_Sea_Moss/boil-sea-moss.jpg', 
    holderImageCollectionCard: 'https://appsformankind-assets.s3.amazonaws.com/Store/Jays_Sea_Moss/gel.jpg', 
    holderImageOrderCard: 'https://appsformankind-assets.s3.amazonaws.com/Store/Jays_Sea_Moss/Irish_Sea_Moss.jpeg', 
    holderUserImage: 'https://appsformankind-assets.s3.amazonaws.com/Store/Jays_Sea_Moss/moss.jpg', 
    
    holderCarouselImageOne: 'https://appsformankind-assets.s3.amazonaws.com/Store/Jays_Sea_Moss/rock.jpg', 
    holderCarouselImageTwo: 'https://appsformankind-assets.s3.amazonaws.com/Store/Jays_Sea_Moss/seaweed.jpg', 
    holderCarouselImageThree:'https://appsformankind-assets.s3.amazonaws.com/Store/Jays_Sea_Moss/waterfall.jpg', 
    holderImage: 'https://appsformankind-assets.s3.amazonaws.com/Store/Jays_Sea_Moss/seamoss.jpg', 
    
  //  mankind_store_id: "DF6598F7",
   // store_id: "DF6598F7", // "P5V8A9ZE",  
    merchant_id: `merchant.com.dhuntleyinc.DevelopingAppsForMankind`,
    publishableKeyLive: "pk_live_51NVCh3JEHBX7gOQA7dW8kmXYPiToufrHpDzRZQ1vhxtUoNXrYXPWtMpELdWetWJhm6Wohk8PXc2ZF1UGfaBOFJl100GZWU6atY",
    publishableKeyTest: "pk_test_51NVCh3JEHBX7gOQAjIjvMGNaHPmbWJov3iFEoLLOUS8oKf9wXowSES3piwS8AB57ml3ejDE1ZNf2tGyzMj7D0GlE007Ms2N4me",
         
}
