// @dhuntleypro/afm-library/src/components/other/auth/login/LoginPopupElements.tsx
// 'use client'

// import styled from 'styled-components';

// // Even Responsive Containers
// export const EvenContainer = styled.div`
//     display: grid;
//     grid-template-columns: repeat(auto-fit, minmax(250px, 2fr));
// `;

// export const EvenContainerWithShadow = styled.div`
//     display: grid;
//     grid-template-columns: repeat(auto-fit, minmax(250px, 2fr));

//     @media screen and (max-width: 768px) {
//         display: flex;
//         flex-direction: column;
//     }

//     /* border-top-right-radius: 20px 20px;
//     border-bottom-right-radius: 20px 20px; */

//     /* box-shadow: 0 8px 32px 0 rgba(255, 255, 255, 0.37); */
//     /* box-shadow: 10px 10px 30px rgba(0, 0, 0, 0.1); */
// `;

// export const EvenContainerContent = styled.div`
//     padding: '20px';
//     background-color: 'white';
//     border-top-right-radius: 30px;
//     border-bottom-right-radius: 30px; 
//     /* padding: 2em; */
//     /* border-radius: 0.3em; */
//     /* background-color: rgba(0, 0, 0, 0.1); */

//         @media screen and (max-width: 768px) {
//              border-bottom-left-radius: 30px;
//         }
// `;


// export const EvenContainerImage = styled.img`
//     max-width: inherit;
//     max-height: inherit;
//     height: inherit;
//     width: inherit;
//     object-fit: cover;
// `;

// // move to library
// export const EvenContainerLeftImage = styled.img`
//     max-width: inherit;
//     max-height: inherit;
//     height: inherit;
//     width: inherit;
//     object-fit: cover;
//     overflow: hidden;
//     /* border-top-left-radius: 20px 20px;
//     border-bottom-left-radius: 20px 20px; */
// `;

// export const EvenContainerImageContainer = styled.div`
//     max-width: 100%;
//     height: 100%;

//     overflow: hidden;
//     border-top-left-radius: 30px;
//     border-bottom-left-radius: 30px;

//     @media screen and (max-width: 768px) {
//         max-height: 170px;
//         width: 100%;
//         overflow: hidden;

//         border-top-left-radius: 30px;
//         border-top-right-radius: 30px;
//         border-bottom-left-radius: 0px;

//         /* padding: 30px; */
//         /* border-radius: 10px; */
//     }
// `;



// export default EvenContainer;
